### DISCUSSION TITLE
Error while trying to use torch.rand_like() on a one dimensional array created using torch.arange()

### DISCUSSION THREAD

**Discussion by VijayV28** [mrdbourke/pytorch-deep-learning#213] [Q&A]
Hey, 
I was trying to implement the torch.rand_like() on an array generated by torch.arange() and got an error:

**Array:**
```python
range1 = torch.arange(start=10, end=101, step=10)
```

**Random function:**
```python
ten_random_numbers = torch.rand_like(input=(range1))
```
This threw an error as shown below:
`RuntimeError: "check_uniform_bounds" not implemented for 'Long'`

However, I later discovered that the `dtype` of the tensor generated by the torch.arange function is `int64`.
The rand_like() works well once I change the `dtype` back to `float32`:

```python
range1 = torch.arange(start=10, end=101, step=10, dtype=torch.float32)
```   

I'd really like to know why doesn't rand_like() support `int64` datatype while the other `tensor_like()` functions do.

**Comment by mrdbourke** [comment]
Hi @VijayV28,

That's an excellent question I don't have the answer to.

I'd like to know too.

Perhaps asking in the PyTorch forums could help?

https://discuss.pytorch.org

**Comment by VijayV28** [comment]
Thanks a lot for your response. Will try your suggestion 👍

**Comment by rezanematpour** [comment]
The torch.rand_like() is specifically designed to sample from a continuous uniform distribution between 0 and 1. The numbers it generates are floating-point values (e.g., 0.112, 0.875, 0.999).
Integer data types like int64 can only store whole numbers (0, 1, 2, ...). Trying to force a fractional value like 0.739 into an integer container creates a logical conflict:
Should it be truncated to 0?
Should it be rounded to 1?
Either choice would destroy the "uniform" nature of the distribution, making the function's behavior mathematically incorrect. Because this operation is ambiguous and misleading, PyTorch disallows it.
